### PR TITLE
Normalise licenses to point to valid SPDX ids

### PR DESCRIPTION
**Pull request by AltGr**
Ref. ocaml/opam#3976
This was done semi-manually. I didn't fix invalid licenses, just
normalised the ones that fit.

### CODE REVIEW

**Comment by thomasblanc** [comment]
So, I think the CI should use xargs :-)

File: packages/conf-linux-libc-dev/conf-linux-libc-dev.0/opam [hunk 1]
```diff
@@ -3,7 +3,7 @@ maintainer: "Markus W. Weissmann <markus.weissmann@in.tum.de>"
 authors: [ "Markus W. Weissmann <markus.weissmann@in.tum.de>" ]
 homepage: "https://kernel.org/"
 bug-reports: "https://github.com/ocaml/opam-repository/issues"
-license: [ "GPL 2.0" ]
+license: [ GPL-2.0-only ]
```
Comment (thomasblanc) [inline packages/conf-linux-libc-dev/conf-linux-libc-dev.0/opam]: You missed the quotes here

**Comment by AltGr** [comment]
Probably not worth the work... but it would be interesting for PRs that _modify_ existing packages if Camelus could do a before/after check and diff that: it would ignore warnings/errors that were already there and you didn't touch, while pointing what you fixed or broke.

 @thomasblanc thanks for spotting it!

**Comment by AltGr** [comment]
I feel sorry for the CI :D
It has a "replay" button, but no "cancel" :/

**Comment by thomasblanc** [comment]
> Probably not worth the work... but it would be interesting for PRs that _modify_ existing packages if Camelus could do a before/after check and diff that: it would ignore warnings/errors that were already there and you didn't touch, while pointing what you fixed or broke.

I disagree. First, because it would add weight on Camelus. Also, and more importantly, it would remove an incentive to maintainers to fix their bad packages. This is only problematic on big prs by maintainers which don't happen every day.

**Comment by AltGr** [comment]
Merging this, besides making licences much cleaner, would allow us to:
- gather some stats on the licenses used (because why not!)
- automatically link to the license text from the package pages on opam.ocaml.org/packages

**Comment by AltGr** [comment]
If there is no objection, let's merge ?

**Comment by kit-ty-kate** [comment]
Thanks a lot for this work

**Pull request merged**